build-system/tasks/helpers.js

```diff
@@ -538,6 +538,7 @@ async function minify(code, map) {
         regex: '_AMP_PRIVATE_$',
         // eslint-disable-next-line google-camelcase/google-camelcase
         keep_quoted: /** @type {'strict'} */ ('strict'),
+        cache: nameCache,
```

Comment:
I checked, and this doesn't actually persist the mangled names. We need to keep the hack.

Reply:
We probably could do it by passing in a subobject to the nameCache parameter.
Something like
```
nameCache: { props: cache }
```

Reply:
Terser uses a `Map` internally, then returns a new object from that map to persist, so it would never update our `cache` object.

Reply:
Lets do it separately though.